File: src/main/java/epicsquid/superiorshields/setup/compat/ThermalCompat.java
package epicsquid.superiorshields.setup.compat;

import com.tterrag.registrate.util.nullness.NonNullFunction;
import epicsquid.superiorshields.item.EnergyShieldItem;
import epicsquid.superiorshields.shield.IEnergyShield;
import net.minecraft.world.item.Item;
import net.minecraftforge.fml.ModList;

public class ThermalCompat {

	private static boolean LOADED;

	public static void init() {
		LOADED = ModList.get().isLoaded("thermal");
	}

	public static NonNullFunction<Item.Properties, EnergyShieldItem> makeThermalShieldOrDefault(IEnergyShield type) {
//		return LOADED ? LoadedOnly.makeThermalShield(type) : props -> new EnergyShieldItem(props, type);
		return props -> new EnergyShieldItem(props, type);
	}

	public static class LoadedOnly {

//		public static NonNullFunction<Item.Properties, EnergyShieldItem> makeThermalShield(IEnergyShield type) {
//			return props -> new ThermalShieldItem(props, type);
//		}
	}
}

File: src/main/kotlin/dev/epicsquid/superiorshields/compat/ThermalCompat.kt
package dev.epicsquid.superiorshields.compat

import dev.epicsquid.superiorshields.item.EnergySuperiorShieldItem
import dev.epicsquid.superiorshields.item.ThermalSuperiorShieldItem
import dev.epicsquid.superiorshields.shield.EnergySuperiorShield
import dev.epicsquid.superiorshields.shield.ThermalSuperiorShield
import net.minecraftforge.fml.ModList

object ThermalCompat {

	private val loaded: Boolean by lazy { ModList.get().isLoaded("thermal") }

	fun thermalShield(
		enchantmentValue: Int,
		type: EnergySuperiorShield,
		maxEnergy: Int,
		barColor: Int
	): EnergySuperiorShieldItem<*> =
		if (loaded) {
			LoadedOnly.thermalShieldBuilder(enchantmentValue, type, maxEnergy, barColor)
		} else {
			EnergySuperiorShieldItem(
				enchantmentValue = enchantmentValue,
				type = type,
				maxEnergy = maxEnergy,
				barColor = barColor
			)
		}

	object LoadedOnly {
		fun thermalShieldBuilder(
			enchantmentValue: Int,
			type: EnergySuperiorShield,
			maxEnergy: Int,
			barColor: Int
		): EnergySuperiorShieldItem<*> =
			ThermalSuperiorShieldItem(
				enchantmentValue = enchantmentValue,
				type = ThermalSuperiorShield(
					type.config
				),
				maxEnergy = maxEnergy,
				barColor = barColor
			)

	}
}